File: PROGRAMMERS/타겟 넘버.java
class Solution {
    
    static int answer;
    static int size;
    
    public int solution(int[] numbers, int target) {
        answer = 0;
        size = numbers.length;
        
        DFS(0, 0, numbers, target);
        
        return answer;
    }
    
    
    private void DFS(int now, int idx, int[] numbers, int target) {
        if(idx == size) {
            if(now == target) answer++;
            
            return;
        }
        
        DFS(now + numbers[idx], idx + 1, numbers, target);
        DFS(now - numbers[idx], idx + 1, numbers, target);
    }
}

File: PROGRAMMERS/타겟 넘버.cpp
#include <bits/stdc++.h>
using namespace std;

int n, k, answer;
vector<int> v;
vector<int> arr;

void DFS(int sum, int len) {
    if(len == n) {
        if(sum == k) answer++;
        return;
    }

    DFS(sum + v[len], len + 1);
    DFS(sum - v[len], len + 1);
}

int solution(vector<int> numbers, int target) {
    n = numbers.size();
    k = target;
    v = numbers;

    DFS(0, 0);

    return answer;
}